airflow/jobs/scheduler_job.py

```diff
@@ -991,10 +1057,13 @@ def is_alive(self, grace_multiplier=None):
         if grace_multiplier is not None:
             # Accept the same behaviour as superclass
             return super().is_alive(grace_multiplier=grace_multiplier)
-        scheduler_health_check_threshold = conf.getint('scheduler', 'scheduler_health_check_threshold')
+        # The object can be retrieved from the database, so it does not contain all the attributes.
+        self.scheduler_health_check_threshold = conf.getint(
```

Comment:
The comment is helpful, but as a result there's no point storing this as an attribute, a local variable makes more sense.

Reply:
Which also means we shouldn't accept it as an argument in the constructor -- it won't ever be used/do anything.

Reply:
The problem is related to the fact that we use the database entities as elements of building the application. Not only as a place to store data.  In most cases, however, the schedulerJob is an application logic object and only in rare cases a database object. I would like to separate it, but to do this, most of the code must be used correctly ie the configuration parameters are passed by the constructor.  If we divide these two aspects in the code, we will be able to make changes much easier and e.g. replace SQLAlchemy with Redis.

Reply:
Combining these two aspects in one class is also hell with cyclical imports.

Reply:
None of that affects that you have created a `scheduler_health_check_threshold` kwarg to the constructor that doesn't ever do anything, but that looks like it does.

Reply:
None of that affects that you have created a `scheduler_health_check_threshold` kwarg to the constructor that doesn't ever do anything, but that looks like it does.